src/iidm/converter/xml/ShuntCompensatorXml.cpp

```diff
@@ -30,33 +31,143 @@ const char* ShuntCompensatorXml::getRootElementName() const {
-    auto bPerSection = context.getReader().getAttributeValue<double>(B_PER_SECTION);
-    auto maximumSectionCount = context.getReader().getAttributeValue<unsigned long>(MAXIMUM_SECTION_COUNT);
-    auto currentSectionCount = context.getReader().getAttributeValue<unsigned long>(CURRENT_SECTION_COUNT);
-    adder.setbPerSection(bPerSection)
-        .setMaximumSectionCount(maximumSectionCount)
-        .setCurrentSectionCount(currentSectionCount);
+void ShuntCompensatorXml::readElement(const std::string& id, ShuntCompensatorAdder& adder, NetworkXmlReaderContext& context) const {
+    IidmXmlUtil::runFromMinimumVersion(IidmXmlVersion::V1_2(), context.getVersion(), [&context, &adder]() {
+        bool voltageRegulatorOn = context.getReader().getOptionalAttributeValue(VOLTAGE_REGULATOR_ON, false);
+        double targetV = context.getReader().getOptionalAttributeValue(TARGET_V, stdcxx::nan());
+        double targetDeadband = context.getReader().getOptionalAttributeValue(TARGET_DEADBAND, stdcxx::nan());
+        adder.setVoltageRegulatorOn(voltageRegulatorOn)
+            .setTargetV(targetV)
+            .setTargetDeadband(targetDeadband);
+    });
+    IidmXmlUtil::runUntilMaximumVersion(IidmXmlVersion::V1_2(), context.getVersion(), [&context, &adder]() {
+        auto bPerSection = context.getReader().getAttributeValue<double>(B_PER_SECTION);
+        auto maximumSectionCount = context.getReader().getAttributeValue<unsigned long>(MAXIMUM_SECTION_COUNT);
+        auto sectionCount = context.getReader().getAttributeValue<unsigned long>(CURRENT_SECTION_COUNT);
+        adder.setSectionCount(sectionCount);
+        adder.newLinearModel()
+            .setBPerSection(bPerSection)
+            .setMaximumSectionCount(maximumSectionCount)
+            .add();
+    });
+    IidmXmlUtil::runFromMinimumVersion(IidmXmlVersion::V1_3(), context.getVersion(), [&context, &adder]() {
+        auto sectionCount = context.getReader().getAttributeValue<unsigned long>(SECTION_COUNT);
+        adder.setSectionCount(sectionCount);
+    });
     readNodeOrBus(adder, context);
+    double p = context.getReader().getOptionalAttributeValue(P, stdcxx::nan());
+    double q = context.getReader().getOptionalAttributeValue(Q, stdcxx::nan());
+    std::string regId;
+    std::string regSide;
+    stdcxx::Properties properties;
+    context.getReader().readUntilEndElement(SHUNT, [&context, &adder, &regId, &regSide, &properties, &id]() {
+        if (context.getReader().getLocalName() == REGULATING_TERMINAL) {
+            regId = context.getAnonymizer().deanonymizeString(context.getReader().getAttributeValue(ID));
+            regSide = context.getReader().getOptionalAttributeValue(SIDE, "");
+        } else if (context.getReader().getLocalName() == PROPERTY) {
+            const std::string& name = context.getReader().getAttributeValue(NAME);
+            const std::string& value = context.getReader().getAttributeValue(VALUE);
+            properties.set(name, value);
```

Comment:
From my point of view, properties are already managed by the base class

Reply:
This method is an overload of the 
```
void AbstractIdentifiableXml<Added, Adder, Parent>::readElement(const std::string& /*id*/, Adder& adder, NetworkXmlReaderContext& context) const {
    Added& identifiable = readRootElementAttributes(adder, context);
    readSubElements(identifiable, context);
}
```
so `readSubElement` is not called anymore and Properties have to be handled manually.
Not sure why, it looks like this is done like that to handle RegulatingTerminal properly (increase scope of read values)

Reply:
Maybe it's not possible to delegate to the parent, because two of the children are needed to build the ShuntCompensator instance.